src/generator.cs

```diff
@@ -3589,9 +3624,9 @@ void GenerateInvoke (bool supercall, MethodInfo mi, MemberInformation minfo, str
 
 	void GenerateNewStyleInvoke (bool supercall, MethodInfo mi, MemberInformation minfo, string selector, string[] args, bool assign_to_temp, Type category_type)
 	{
-		bool arm_stret = Stret.ArmNeedStret (mi);
-		bool x86_stret = Stret.X86NeedStret (mi);
-		bool x64_stret = Stret.X86_64NeedStret (mi);
+		bool arm_stret = Stret.ArmNeedStret (mi.ReturnType);
+		bool x86_stret = Stret.X86NeedStret (mi.ReturnType);
+		bool x64_stret = Stret.X86_64NeedStret (mi.ReturnType);
```

Comment:
minor, but `ReturnType` is a virtual property that we (but not the compiler/JIT/AOT) know won't change between calls. Let's make it a variable.